ui/lib/core/addon/mixins/replication-actions.js

```diff
@@ -88,16 +88,10 @@ export default Mixin.create({
     if (action === 'disable') {
       yield this.onDisable();
     }
-    if (mode === 'secondary' && replicationMode === 'dr') {
-      // return mode so you can properly handle the transition
-      return mode;
-    }
     if (action === 'enable') {
-      try {
-        yield this.onEnable(replicationMode); // should not be called for dr secondary
-      } catch (e) {
-        // TODO handle error
-      }
+      /// onEnable is a method available only to route vault.cluster.replication.index
+      // if action 'enable' is called from vault.cluster.replication.mode.index this method is not called
```

Comment:
when you say 'this method is not called', do you mean that we never hit this conditional and execute line 94 if we're on the vault.cluster.replication.mode.index?

Reply:
We hit the conditional, but because it can't find the method 'onEnable' it just stops.   There is some refactoring here that could help, but onEnable is used by a lot of the actions I wasn't dealing with so I just added a comment as it's a gotchya when trying to problem solve.